src/components/ebay-calendar/index.marko

```diff
@@ -0,0 +1,110 @@
+                <ebay-icon-button
+                    transparent size="small"
+                    disabled=(state.disableBefore && component.getFirstVisibleISO() <= state.disableBefore)
+                    aria-label=`Show ${component.monthTitle(component.getMonthDate(state.offset - 1))}`
+                    onClick('prevMonth')>
+                    <ebay-chevron-left-icon/>
+                </ebay-icon-button>
+                <for|monthDate| of=monthDates>
+                    <h3>${component.monthTitle(monthDate)}</h3>
+                </for>
+                <ebay-icon-button
+                    transparent size="small"
+                    disabled=(state.disableAfter && component.getLastVisibleISO() >= state.disableAfter)
+                    aria-label=`Show ${component.monthTitle(component.getMonthDate(state.offset + numMonths))}`
+                    onClick('nextMonth')>
+                    <ebay-chevron-right-icon/>
+                </ebay-icon-button>
+            </div>
+        </div>
+    </if>
+    <div.calendar__body>
+        <for|monthDate| of=monthDates>
+            <div.calendar__month>
+                <table onKeydown('onKeyDown')>
+                    <caption>${component.monthTitle(monthDate)}</caption>
+                    <thead>
+                        <tr>
+                            <for|dayName| of=state.weekdayLabels>
+                                <th>${dayName}</th>
+                            </for>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        $ const numBlankDays = (monthDate.getDay() - state.firstDayOfWeek + 7) % 7;
+                        $ const year = monthDate.getFullYear();
+                        $ const month = monthDate.getMonth();
+                        $ const daysInMonth = new Date(year, month + 1, 0).getDate();
+                        <for|row| from=0 to=Math.floor((numBlankDays + daysInMonth) / 7)>
+                            <tr>
+                                $ let startDate = row * 7 - numBlankDays + 1;
+                                $ let endDate = startDate + 6;
+                                <if(row === 0 && numBlankDays !== 0)>
+                                    <td colspan=numBlankDays/>
+                                    $ startDate = 1;
+                                </if>
+                                <else-if(endDate > daysInMonth)>
+                                    $ endDate = daysInMonth;
+                                </else-if>
+                                <for|day| from=startDate to=endDate>
+                                    $ const dayISO = toISO(new Date(year, month, day));
+                                    $ const isRangeStart = dayISO === state.rangeStart;
+                                    $ const isInRange = (state.rangeStart && state.rangeEnd
+                                                && dayISO >= state.rangeStart && dayISO <= state.rangeEnd);
```

Comment:
This logic is a bit confusing and could possibly be very brittle. 
Having `&&` mixed with `>` like this is hard to understand. 
Could be better in a common method in the component where it has each of these statements line by line.

Reply:
+1 to what Andrew said. Ideally this can be extracted to a child component `ebay-calendar-month` or something similar.

Reply:
Ok, I moved it into the component. Hopefully that's a little bit more clear.